packages/edit-site/src/hooks/use-theme-style-variations/use-theme-style-variations-by-property.js

```diff
@@ -47,17 +47,31 @@ export function removePropertyFromObject( object, property ) {
 	return object;
 }
 
+/**
+ * Checks whether a style variation is empty.
+ *
+ * @param {Object} variation          A style variation object.
+ * @param {string} variation.title    The title of the variation.
+ * @param {Object} variation.settings The settings of the variation.
+ * @param {Object} variation.styles   The styles of the variation.
+ * @return {boolean} Whether the variation is empty.
+ */
+function isEmptyStyleVariation( { title, settings, styles } ) {
+	return (
+		title === __( 'Default' ) || // Always preseve the default variation.
+		Object.keys( settings ).length > 0 ||
+		Object.keys( styles ).length > 0
```

Comment:
I'm not familiar with how `theme.json` settings get applied. From looking at the  structure, I'm wondering if this should be an AND:

`( Object.keys( settings ).length > 0 && Object.keys( styles ).length > 0 )`

It looks like it could be a valid theme style while having a setting or style? Or does it need both?

Reply:
Ah the problem here is actually the name of the function. It is returning true if the variation is NOT empty!